frontend/src/lib/components/BranchButton.svelte

```diff
@@ -0,0 +1,431 @@
+			credentials: 'include',
+			headers: {
+				'Content-Type': 'application/json'
+			}
+		});
+
+		// Check if the response is successful
+		if (!response.ok) {
+			const errorMessage = await response.json();
+			console.error('Failed to fetch branches:', errorMessage);
+			addToast({
+				message: `Error fetching branches: ${response.statusText}. ${JSON.stringify(errorMessage)}`,
+				type: ToastType.Error,
+				dismissible: true
+			});
+			return { nonProtectedBranches: [], protectedBranches: [] };
+		}
+
+		// Extract and set the branches if the response is successful
+		const data = await response.json();
+		const branches: string[] = data.data?.branches || [];
+
+		// Map through branches to create Branch objects
+		const nonProtectedBranches: Branch[] = branches
+			.filter((branch: string) => !branch.includes('(protected)'))
+			.map((branch: string) => ({
+				name: branch.split(' (')[0],
+				isProtected: false
+			}));
+
+		const protectedBranches: Branch[] = branches
+			.filter((branch: string) => branch.includes('(protected)'))
+			.map((branch: string) => ({
+				name: branch.split(' (')[0],
+				isProtected: true
+			}));
+
+		return { nonProtectedBranches, protectedBranches };
+	}
+
+	onMount(async () => {
+		const { nonProtectedBranches: fetchedNonProtected, protectedBranches: fetchedProtected } =
+			await fetchExistingBranches();
+		nonProtectedBranches = fetchedNonProtected;
+		protectedBranches = fetchedProtected;
+	});
+
+	async function setBranchName(input: string) {
+		if (!input) return;
+
+		// Define validation rules
+		const maxLength = 255; // Maximum length for branch name
+		const invalidCharacters = /[~^:?*<>|]/; // Invalid special characters
+		const startsWithLetterOrNumber = /^[a-zA-Z0-9]/; // Starts with letter or number
+		const containsSpaces = /\s/; // Contains spaces
+
+		// Define an array of rules
+		const rules = [
+			'Branch names must start with a letter or a number.',
+			'Branch names cannot contain spaces. Use dashes (-) or underscores (_) instead.',
+			'Branch names cannot contain special characters like ~, ^, :, ?, *, and others.',
+			`Branch names must be shorter than ${maxLength} characters.`
+		];
+
+		const isValidBranchName = (name: string) => {
+			return (
+				!containsSpaces.test(name) &&
+				startsWithLetterOrNumber.test(name) &&
+				!invalidCharacters.test(name) &&
+				name.length <= maxLength
+			);
+		};
+
+		// Validate branch name
+		if (!isValidBranchName(input)) {
+			addToast({
+				message: 'Please ensure your branch name follows these rules:\n' + rules.join('\n'),
+				type: ToastType.Warning,
+				dismissible: true
+			});
+			return;
+		}
+
+		if (protectedBranches.some((branch) => branch.name === input)) {
+			addToast({
+				message:
+					'Please select an existing branch name from the list of non-protected branches.\n You can also create your own',
+				type: ToastType.Warning,
+				dismissible: true,
+				timeout: 1500
+			});
+			return;
+		}
+
```

Comment:
So for this, I believe it would make for a more intuitive UI if you added the `disabled` attribute to every protected branch button.

See <https://www.w3schools.com/tags/att_button_disabled.asp> for more